lib/pure/asyncdispatch.nim

```diff
@@ -123,15 +123,26 @@
 ##   if future.failed:
 ##     # Handle exception
 ##
-##
-## Discarding futures
+## Handling futures
 ## ==================
 ##
-## Futures should **never** be discarded. This is because they may contain
-## errors. If you do not care for the result of a Future then you should
-## use the `asyncCheck` procedure instead of the `discard` keyword. Note
-## however that this does not wait for completion, and you should use
-## `waitFor` for that purpose.
+## Futures should **never** be discarded directly because they may contain
+## errors. Instead, use `asyncCheck`, `waitFor`, or `await`, e.g., `waitFor
+## someAsyncProc()`. See the below table for when to use each:
+##
+## ================  =====================   ================
+## Procedure         Context                 Blocking
+## ================  =====================   ================
+## `asyncCheck`      non-async and async     non-blocking
+## `waitFor`         non-async               blocking*
+## `await`           async                   blocking*
```

Comment:
`await` definitely isn't "blocking". I think what we really want to say here is that "`waitFor` blocks current thread" and "`await` suspends current proc".

Reply:
So something like this, and then maybe remove lines 141-142
```nim
## `asyncCheck`    non-async and async     non-blocking
## `waitFor`       non-async               blocks current thread
## `await`         async                   suspends current procedure
```